Terraform/Home_Task_2/modules/networking/main.tf

```diff
@@ -0,0 +1,183 @@
+resource "aws_vpc" "vpc_1" {
+  cidr_block       = var.vpc_1_cidr_block
+  instance_tenancy = var.vpc_1_instance_tenancy
+}
+
+resource "aws_internet_gateway" "igw" {
+  vpc_id = aws_vpc.vpc_1.id
+}
+
+resource "aws_subnet" "public_subnet_1" {
+  vpc_id                  = aws_vpc.vpc_1.id
+  cidr_block              = var.public_subnet_1_cidr_block
+  availability_zone       = var.public_subnet_1_availability_zone
+  map_public_ip_on_launch = var.public_subnet_1_map_public_ip_on_launch
+}
+resource "aws_subnet" "public_subnet_2" {
+  vpc_id                  = aws_vpc.vpc_1.id
+  cidr_block              = var.public_subnet_2_cidr_block
+  availability_zone       = var.public_subnet_2_availability_zone
+  map_public_ip_on_launch = var.public_subnet_1_map_public_ip_on_launch
+}
+
+resource "aws_subnet" "private_subnet_1" {
+  vpc_id            = aws_vpc.vpc_1.id
+  cidr_block        = var.private_subnet_1_cidr_block
+  availability_zone = var.private_subnet_1_availability_zone
+}
+resource "aws_subnet" "private_subnet_2" {
+  vpc_id            = aws_vpc.vpc_1.id
+  cidr_block        = var.private_subnet_2_cidr_block
+  availability_zone = var.private_subnet_2_availability_zone
+}
+
+resource "aws_nat_gateway" "nat_gtw_1" {
+  allocation_id = aws_eip.eip_1.id
+  subnet_id     = aws_subnet.public_subnet_1.id
+
+  tags = merge({ Name = "GW_NAT_1" }, var.general_tags)
+
+  depends_on = [aws_internet_gateway.igw]
+}
+
+resource "aws_eip" "eip_1" {
+  vpc        = var.eip_vpc

```

Comment:
delete vpc and depends_on params
leave just a clean block: resource "aws_eip" "eip_1" {}